kotlinx-coroutines-core/common/src/flow/Flow.kt

```diff
@@ -149,8 +149,8 @@ import kotlin.coroutines.*
  * it hard to reason about the code because an exception in the `collect { ... }` could be somehow "caught"
  * by an upstream flow, limiting the ability of local reasoning about the code.
  *
- * Currently, the flow infrastructure does not enforce exception transparency contracts, however, it might be enforced
- * in the future either at run time or at compile time.
+ * Flow machinery enforces exception transparency at runtime an throws [IllegalStateException] on any attempt to emit value,
+ * if an exception has been thrown on previous attemp.
```

Comment:
Typos: 
* "_an_ throws" -> "and throws"
* "on previous _attemp_" -> "on a previous attempt" (also article)